src/api/hooks/useGetProductsDetail.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+import { fetchData } from '@/api/hooks/getAPI';
+import type { DetailData, ProductDetailParams } from '@/types';
+
+export default function useGetProductsDetail({ id }: ProductDetailParams) {
+  const [data, setData] = useState<DetailData>();
+  const [isLoading, setLoading] = useState(true);
+  const [isError, setError] = useState(false);
+
+  useEffect(() => {
+    const fetchProducts = async () => {
+      try {
+        setLoading(true);
+        setError(false);
+        const response = await fetchData(`/api/v1/products/${id}/detail`);
+
+        setData(response.detail);
+        setLoading(false);
+      } catch {
+        setError(true);
+        setData(undefined);
+      }
```

Comment:
fetchData가 정상동작으로 종료되었을 경우 뿐아니라 
에러발생 시에도 loading 종료 처리를 해주어야 할 것 같습니다.

```suggestion
      try {
        setLoading(true);
        setError(false);
        const response = await fetchData(`/api/v1/products/${id}/detail`);

        setData(response.detail);
      } catch {
        setError(true);
        setData(undefined);
      } finally {
        setLoading(false);
      }
```